src/webgpu/shader/execution/memory_model/coherence.spec.ts

```diff
@@ -54,6 +52,22 @@ const storageMemoryCorrTestCode = `
   atomicStore(&results.value[id_1].r1, r1);
 `;
 
+const workgroupStorageMemoryCorrTestCode = `
+  atomicStore(&test_locations.value[x_0], 1u);
+  let r0 = atomicLoad(&test_locations.value[x_1]);
+  let r1 = atomicLoad(&test_locations.value[y_1]);
+  atomicStore(&results.value[shuffled_workgroup * workgroupXSize + id_1].r0, r0);
+  atomicStore(&results.value[shuffled_workgroup * workgroupXSize + id_1].r1, r1);
+`;
+
+const corrRMWTestCode = `
+  atomicExchange(&test_locations.value[x_0], 1u);
+  let r0 = atomicLoad(&test_locations.value[x_1]);
+  let r1 = atomicAdd(&test_locations.value[y_1], 0u);
```

Comment:
So we have two independent dimensions:    storage vs. workgroup; and  read-read vs read-RMW.
So there should be 4 variants, but there are only 3.

Seems we should also test RMW on the workgroup storage.

Reply:
For consistency with the others, should this be called storageMemoryCorrRMWTestCode ?